packages/core/src/telemetry/utility.ts

```diff
@@ -1,10 +1,17 @@
 import { propagation, trace } from '@opentelemetry/api';
 import type { Context } from '@opentelemetry/api';
 
-// Helper function to check if telemetry is active
+/**
+ * Check if telemetry is active and enabled.
+ * This function checks both if telemetry is enabled in the configuration AND if a tracer exists.
+ *
+ * @param tracerName - Optional tracer name to check (currently unused, kept for backwards compatibility)
+ * @returns true if telemetry is both initialized and enabled, false otherwise
+ */
 export function hasActiveTelemetry(tracerName: string = 'default-tracer'): boolean {
   try {
-    return !!trace.getTracer(tracerName);
+    const isEnabled = globalThis.__TELEMETRY__?.isEnabled() ?? true;
+    return isEnabled && !!trace.getTracer(tracerName);
```

Comment:
Actually, `trace.getTracer` is always going to return a Tracer instance, and hence `!!trace.getTracer(tracerName)` always evaluates to true, regardless of the value of `tracerName` passed-in.

The none-default call, is only made by the `InstrumentClass` decorator, in particular when decorating an `Agent`. We could change the signature of `hasActiveTelemetry` and made the caller rely only on the value of the `telemetry.enabled` property, If that's preferred. The effect will be the same, because what comes after the `&&` is always true.